docs/source/export.rst

```diff
@@ -0,0 +1,113 @@
+
+.. code-block:: python
+
+   with NWBHDF5IO(self.read_path, mode='r') as read_io:
+       with NWBHDF5IO(self.export_path, mode='w') as export_io:
+           export_io.export(src_io=read_io)
+
+FAQ
+---
+
+Can I read an `NWBFile` from disk, modify it, and then export the modified `NWBFile`?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Yes, you can export the in-memory `NWBFile` after modifying it in memory. The modifications will appear in the exported
+file and not the read file.
+
+- If the modifications are removals or additions of containers, then no special action must be taken, as long as the
+  `NWBFile` hierarchy is updated correspondingly.
+- If the modifications are changes to attributes, then
+  :py:meth:`Container.set_modified() <hdmf.container.AbstractContainer.set_modified>` must be called
+  on the `NWBFile` before exporting.
+
+.. code-block:: python
+
+   with NWBHDF5IO(self.read_path, mode='r') as read_io:
+       nwbfile = read_io.read()
+       # ...  # modify nwbfile
+       nwbfile.set_modified()  # this may be necessary if the modifications are changes to attributes
+       with NWBHDF5IO(self.export_path, mode='w') as export_io:
+           export_io.export(src_io=read_io, nwbfile=nwbfile)
+
+.. note::
+
+  Modifications to :py:class:`h5py.Dataset <h5py.Dataset>` objects act *directly* on the read file on disk.
+  Changes are applied immediately and do not require exporting or writing the file. If you want to modify a dataset
+  only in the new file, than you should replace the whole object with a new array holding the modified data. To
+  prevent unintentional changes to the source file, the source file should be opened with ``mode='r'``.
+
+Can I export a newly instantiated `NWBFile`?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+No, you can only export containers that have been read from a file. The ``src_io`` argument is required in
+:py:meth:`NWBHDF5IO.export <pynwb.NWBHDF5IO.export>`.
+
+Can I read a `NWBFile` from disk and export only part of the `NWBFile`?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+It depends. You can only export the root `NWBFile` from a file. To export the root `NWBFile` without certain other
+sub-containers in the hierarchy, you can remove those other containers before exporting. However, you cannot export
+only a sub-container of the `NWBFile` hierarchy.
+
+Can I write a newly instantiated `NWBFile` to two different files?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+PyNWB does not allow you to write an `NWBFile` that was not read from a file to two different files. For example, if you
+instantiate `NWBFile` A and write it file 1 and then try to write it to file 2, an error will be raised. However, you
+can read `NWBFile` A from file 1 and then export it to file 2, with or without modifications to `NWBFile` A in
+memory.
```

Comment:
I think it would be useful to have a short code example to illustrate the basic steps for this.

Reply:
Done